src/DI/ContainerBuilder.php

```diff
@@ -643,9 +664,12 @@ private function generateService($name)
 			->addParameter('container')
 				->setTypeHint($this->className);
 
-		$factoryClass->addMethod($def->getImplementType())
+		$method = $factoryClass->addMethod($def->getImplementType())
 			->setParameters($this->convertParameters($def->parameters))
 			->setBody(str_replace('$this', '$this->container', $code));
+		if (PHP_VERSION_ID >= 70000) {
```

Comment:
Yeah, this looks way better. :) There is one small problem with the return type adding which was pointed out to me on chat yesterday - it currently adds the return type declaration even if it's not specified in interface. 

It doesn't break applications (and tests) because PHP allows adding the return type in any point of hierarchy, but still changes how the FW behaved up until now - is that considered a BC break (and a issue in general), since it affects only automatically generated classes and doesn't break anything?